stdlib/universe/typeconv_test.go

```diff
@@ -2,51 +2,505 @@ package universe
+		},
+		{
+			name: "string(v:6)",
+			v:    values.Time(1136239445999999999),
+			want: "2006-01-02T22:04:05.999999999Z",
+			expectErr: errors.New("Cannot convert."),
+		},
+		{
+			name: "string(v:7)",
+			v:    values.Duration(184000000000),
+			want: "3m4s",
+			expectErr: errors.New("Cannot convert."),
+		},
+		{
+			name: "string(v:8)",
+			v:    int64(-541),
+			want: "-541",
+			expectErr: errors.New("Cannot convert."),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			myMap := map[string]values.Value{
+				"v": values.New(tc.v),
+			}
+			args := values.NewObjectWithValues(myMap)
+			c := &stringConv{}
+			got, err := c.Call(args)
+			if err != nil {
+				if want, got := tc.expectErr.Error(), err.Error(); got != want {
+					t.Errorf("unexpected error - want: %s, got: %s", want, got)
+				}
+				return
+			}
+			want := values.NewString(tc.want)
+			if !got.Equal(want) {
+				t.Errorf("Wanted: %s, got: %v", want, got)
+			}
+		})
+	}
+}
+
+func TestTypeconv_Int(t *testing.T) {
 	testCases := []struct {
-		name   string
-		v      interface{}
-		want   bool
-	} {
-		{ 
-			name : "bool(v:1)",
-			v : int64(1),
-			want : true,
+		name string
+		v    interface{}
+		want int64
+		expectErr error
+	}{
+		{
+			name: "int64(v:1)",
+			v:    "4615",
+			want: int64(4615),
+			expectErr: errors.New("Cannot convert."),
+		},
+		{
+			name: "int64(v:2)",
+			v:    uint64(123),
+			want: int64(123),
+			expectErr: errors.New("Cannot convert."),
+		},
+		{
+			name: "int64(v:3)",
+			v:    float64(-728),
+			want: int64(-728),
+			expectErr: errors.New("Cannot convert."),
 		},
 		{
-			name : "bool(v:2)",
-			v : "true",
-			want : true,
+			name: "int64(v:4)",
+			v:    true,
+			want: int64(1),
+			expectErr: errors.New("Cannot convert."),
 		},
 		{
-			name : "bool(v:3)",
-			v : uint64(1),
-			want : true,
+			name: "int64(v:4)",
+			v:    false,
+			want: int64(0),
+			expectErr: errors.New("Cannot convert."),
 		},
 		{
-			name : "bool(v:4)", 
-			v : float64(1),
-			want : true,
+			name: "int64(v:5)",
+			v:    values.Time(1136239445999999999),
+			want: int64(1136239445999999999),
+			expectErr: errors.New("Cannot convert."),
+		},
+		{
+			name: "int64(v:6)",
+			v:    values.Duration(123456789),
+			want: int64(123456789),
+			expectErr: errors.New("Cannot convert."),
+		},
+		{
+			name : "int64(error)",
+			v: "notanumber",
+			want: 0,
+			expectErr: errors.New("strconv.ParseInt: parsing \"notanumber\": invalid syntax"),
 		},
 	}
 	for _, tc := range testCases {
-		t.Run(tc.name, func(t*testing.T) {
-			myMap := map[string]values.Value {
-				"v" : values.New(tc.v),
+		t.Run(tc.name, func(t *testing.T) {
+			myMap := map[string]values.Value{
+				"v": values.New(tc.v),
+			}
+			args := values.NewObjectWithValues(myMap)
+			c := &intConv{}
+			got, err := c.Call(args)
+			if err != nil {
+				if want, got := tc.expectErr.Error(), err.Error(); got != want {
```

Comment:
around this add a check, `if tc.expectErr != nil  { ... }  else {t.Errorf("unexpected error %s", err.Error)}

Also in all test cases if you are not expecting an error in that case, then you should not have anything in there.